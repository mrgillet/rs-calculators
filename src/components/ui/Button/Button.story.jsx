import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import Button from './Button';

storiesOf('UI/Button', module)
  .addDecorator(centered)
  .add('Default white theme', () => <Button>Default white button</Button>)
  .add('Blue theme', () => <Button theme="blue">Primary blue button</Button>);
