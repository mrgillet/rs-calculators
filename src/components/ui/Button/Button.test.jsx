import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

let clickTestValue = 0;
const inCreaseValueCount = () => {
  clickTestValue += 1;
};

test('UI/Button', () => {
  const wrapper = shallow(<Button theme="blue" onClick={inCreaseValueCount}>Hello</Button>);
  wrapper.simulate('click');

  expect(wrapper.text()).toEqual('Hello');
  expect(clickTestValue).toEqual(1);

  expect(wrapper).toMatchSnapshot();
});
