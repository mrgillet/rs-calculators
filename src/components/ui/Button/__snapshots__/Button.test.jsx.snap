// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Button 1`] = `
<button
  className="button button--blue"
  onClick={[Function]}
>
  Hello
</button>
`;
